modules/default/compliments/compliments.js

```diff
@@ -91,20 +123,49 @@ Module.register("compliments", {
 		// Add compliments based on weather
 		if (this.currentWeatherType in this.config.compliments) {
 			Array.prototype.push.apply(compliments, this.config.compliments[this.currentWeatherType]);
+			// if the predefine list doesn't include it (yet)
+			if (!this.pre_defined_types.includes(this.currentWeatherType)) {
+				// add it
+				this.pre_defined_types.push(this.currentWeatherType);
+			}
 		}
 
 		// Add compliments for anytime
 		Array.prototype.push.apply(compliments, this.config.compliments.anytime);
 
+		// get the list of just date entry keys
+		let temp_list = Object.keys(this.config.compliments).filter((k) => {
+			if (this.pre_defined_types.includes(k)) return false;
+			else return true;
+		});
+
+		let date_compliments = [];
 		// Add compliments for special days
-		for (let entry in this.config.compliments) {
-			if (new RegExp(entry).test(date)) {
-				// Only display compliments configured for the day if specialDayUnique is set to true
-				if (this.config.specialDayUnique) {
-					compliments.length = 0;
-				}
-				Array.prototype.push.apply(compliments, this.config.compliments[entry]);
+		for (let entry of temp_list) {
+			// check if this could be a cron type entry
+			if (entry.includes(" ")) {
```

Comment:
we got a m.includes(" ") some lines above. Maybe just make this a function
checkCronEntry 
that does this. Make it clearer and gets rid of the comments

Reply:
ok